src/wallet/messaging_daemon.h

```diff
@@ -0,0 +1,131 @@
+    "message": "SXMgdGhpcyBmYXN0ZXIgbm93Pw==",
+    "fromAddress": "BM-2cXwZ8TEHK3fGqm3DKAUAk3nfDjmTawZr6",
+    "receivedTime": "1529832273",
+    "subject": "U2Vjb25kIG1lc3NhZ2U="
+}]}
+*/
+
+namespace tools
+{
+  namespace bitmessage_rpc
+  {
+    
+    struct message_info
+    {
+      uint32_t encodingType;
+      std::string toAddress;
+      uint32_t read;
+      std::string msgid;
+      std::string message;
+      std::string fromAddress;
+      std::string receivedTime;
+      std::string subject;
+      
+      BEGIN_KV_SERIALIZE_MAP()
+        KV_SERIALIZE(encodingType)
+        KV_SERIALIZE(toAddress)
+        KV_SERIALIZE(read)
+        KV_SERIALIZE(msgid)
+        KV_SERIALIZE(message);
+        KV_SERIALIZE(fromAddress)
+        KV_SERIALIZE(receivedTime)
+        KV_SERIALIZE(subject)
+      END_KV_SERIALIZE_MAP()
+    };
+    
+    struct inbox_messages_response
+    {
+      std::vector<message_info> inboxMessages;
+
+      BEGIN_KV_SERIALIZE_MAP()
+        KV_SERIALIZE(inboxMessages)
+      END_KV_SERIALIZE_MAP()
+    };
+    
+  }
+}
+
+namespace tools
+{
+  class messaging_daemon: public epee::http_server_impl_base<messaging_daemon>
+  {
+  public:
+    typedef epee::net_utils::connection_context_base connection_context;
+
+    static const char* tr(const char* str);
+
+    messaging_daemon();
+    ~messaging_daemon();
+
+    bool init(const boost::program_options::variables_map *vm);
+    bool run();
+    void stop();
+
+  private:
+    CHAIN_HTTP_TO_MAP2(connection_context);
+
+    BEGIN_URI_MAP2()
+      BEGIN_JSON_RPC_MAP("/json_rpc")
+        MAP_JON_RPC_WE("add", on_add, messaging_daemon_rpc::COMMAND_RPC_ADD)
+        MAP_JON_RPC_WE("send_message", on_send_message, messaging_daemon_rpc::COMMAND_RPC_SEND_MESSAGE)
+        MAP_JON_RPC_WE("receive_messages", on_receive_messages, messaging_daemon_rpc::COMMAND_RPC_RECEIVE_MESSAGES)
+        MAP_JON_RPC_WE("delete_message", on_delete_message, messaging_daemon_rpc::COMMAND_RPC_DELETE_MESSAGE)
+      END_JSON_RPC_MAP()
+    END_URI_MAP2()
+
+    bool on_add(const messaging_daemon_rpc::COMMAND_RPC_ADD::request& req, messaging_daemon_rpc::COMMAND_RPC_ADD::response& res, epee::json_rpc::error& er);
+    bool on_send_message(const messaging_daemon_rpc::COMMAND_RPC_SEND_MESSAGE::request& req, messaging_daemon_rpc::COMMAND_RPC_SEND_MESSAGE::response& res, epee::json_rpc::error& er);
+    bool on_receive_messages(const messaging_daemon_rpc::COMMAND_RPC_RECEIVE_MESSAGES::request& req, messaging_daemon_rpc::COMMAND_RPC_RECEIVE_MESSAGES::response& res, epee::json_rpc::error& er);
+    bool on_delete_message(const messaging_daemon_rpc::COMMAND_RPC_DELETE_MESSAGE::request& req, messaging_daemon_rpc::COMMAND_RPC_DELETE_MESSAGE::response& res, epee::json_rpc::error& er);
+
+    void handle_rpc_exception(const std::exception_ptr& e, epee::json_rpc::error& er, int default_error_code);
+
+    const boost::program_options::variables_map *m_vm;
```

Comment:
I think there is no need to make it class member because it's used only in one function